homeassistant/components/myuplink/sensor.py

```diff
@@ -102,4 +164,62 @@ def __init__(
     def native_value(self) -> StateType:
         """Sensor state value."""
         device_point = self.coordinator.data.points[self.device_id][self.point_id]
+        if device_point.value == MARKER_FOR_UNKNOWN_VALUE:
+            return None
         return device_point.value  # type: ignore[no-any-return]
+
+
+class MyUplinkEnumSensor(MyUplinkDevicePointSensor):
+    """Representation of a myUplink device point sensor for ENUM device_class."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        device_point: DevicePoint,
+        entity_description: SensorEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            device_point=device_point,
+            entity_description=entity_description,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        self._attr_options = [x["text"].capitalize() for x in device_point.enum_values]
+        self.options_map = {x["value"]: x["text"] for x in device_point.enum_values}
+
+    @property
+    def native_value(self) -> str:
+        """Sensor state value for enum sensor."""
+        device_point = self.coordinator.data.points[self.device_id][self.point_id]
+        return self.options_map[str(int(device_point.value))].capitalize()  # type: ignore[no-any-return]
```

Comment:
Why is the option map not capitalized by default?

Reply:
Good catch!
It just a result of moving things around. There is no reason for doing it on every value update.